Added possibility to configure and add toolbar buttons

bolelamx opened the pull request:
Example of configuring and using:

```
angular.module('myapp').config(['ngWigToolbarProvider', function(ngWigToolbarProvider) {
        ngWigToolbarProvider.setConfig({
            button: ['list1', 'list2', 'bold', 'italic', 'link'],
            format: ['p', 'h1', 'h2', 'h3'],
            other: ['source']
        });
}]);
```

Example of adding custom buttons:

```
angular.module('myapp').config(['ngWigToolbarProvider', function(ngWigToolbarProvider) {        
        ngWigToolbarProvider.addButton('underline',{title: 'Underline', command: 'underline', styleClass: 'nw-button--underline'});
	ngWigToolbarProvider.addButton('strikethrough');
}]);
```

stevermeister commented on src/javascript/app/ng-wig/ngWigDirective.js:
```diff
@@ -1,5 +1,5 @@
 angular.module('ngWig')
-  .directive('ngWig', function (ngWigToolbarProvider) {
```
:+1:

stevermeister commented:
could you add example to actual code, please?

stevermeister commented on src/javascript/app/ng-wig/ngWigToolbarProvider.js:
```diff
@@ -1,29 +1,96 @@
+        addToConfig(BUTTON_TYPE, name);
+    };
+
+    this.addFormat = function (name, value) {
```
format dropdown would be plugin itself so no need to put it here

stevermeister commented on src/javascript/app/ng-wig/ngWigToolbarProvider.js:
```diff
@@ -1,29 +1,96 @@
+          italic: {title: 'Italic', command: 'italic', styleClass: 'nw-button--italic'},
+          link: {title: 'Link', command: 'createlink', styleClass: 'nw-button--link'}
+      },
+      format: {
```
commented behind

stevermeister commented on src/javascript/app/ng-wig/ngWigToolbarProvider.js:
```diff
@@ -1,29 +1,96 @@
-        });
-        return toolbarButtons;
+angular.module('ngWig').provider('ngWigToolbar', function () {
+    var FORMAT_TYPE = 'format',
```
didn't get the idea of this variables

bolelamx commented:
I removed format dropdow from toolbar provider and added examples of configuring and adding custom button to code. Please note that also need to add css styles for button icon.
